app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/btle/AbstractTransaction.java

```diff
@@ -0,0 +1,49 @@
+/*  Copyright (C) 2015-2019 Andreas Shimokawa, Carsten Pfeiffer, Andreas Boehler
+
+    This file is part of Gadgetbridge.
+
+    Gadgetbridge is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Gadgetbridge is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>. */
+package nodomain.freeyourgadget.gadgetbridge.service.btle;
+
+import java.text.DateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+public abstract class AbstractTransaction {
+    private final String mName;
+    private final long creationTimestamp = System.currentTimeMillis();
+
+
+    public AbstractTransaction(String taskName) {
+        this.mName = taskName;
+    }
+
+    public String getTaskName() {
+        return mName;
+    }
+
+    protected String getCreationTime() {
+        return DateFormat.getTimeInstance(DateFormat.MEDIUM).format(new Date(creationTimestamp));
+    }
+
+    public int getActionSize() {
+        return 0;
```

Comment:
This method should rather be abstract (and named getActionCount(), but that might just be my personal preference :-)

Apart from that: :+1: :+1: